crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -231,6 +234,16 @@ impl DiskFileBlobStoreInner {
         Ok(self.blob_disk_file(tx).is_file())
     }
 
+    /// Returns all the blob transactions which are in the cache or on the disk.
+    fn retain_existing(&self, txs: Vec<B256>) -> Result<Vec<B256>, BlobStoreError> {
+        let mut cache = self.blob_cache.lock();
+        Ok(txs
+            .iter()
+            .filter(|&tx| cache.get(tx).is_some() || self.blob_disk_file(*tx).is_file())
+            .cloned()
+            .collect())
```

Comment:
hmm, this is a bit more tricky then I thought.

what we should actually do here is 
a first over over the cache with

https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.partition

then drop the cache lock
and then a second pass over those that weren't in the cache via `self.blob_disk_file(*tx).is_file()`

because we don't wont to hold the cache lock while we do IO

Reply:
@mattsse ptal, thanks